### ISSUE TITLE
Is it possible to resize PiP?

### ISSUE THREAD

**Issue by fromA-Z** [GoogleChromeLabs/picture-in-picture-chrome-extension#43]
It's a bit small for certain use cases. Is there any way to resize it?

**Comment by beaufortfrancois** [comment]
Can you share a screenshot of where it is a bit small?
Only user can resize pip window, extension can't do that.

**Comment by fromA-Z** [comment]
<img width="1280" alt="Screen Shot 2020-03-31 at 6 02 48 AM" src="https://user-images.githubusercontent.com/30534360/78014105-73a63000-7315-11ea-9a5a-2dc1064b858e.png">

How can I resize the PiP window?

**Comment by beaufortfrancois** [comment]
The same way you would resize a native OS window. Drag its corner.

**Comment by isaquepereira** [comment]
The resize has a limit, no? Can you remove or set a bigger limit?
I want use it for my code studies, but looks so tiny... See:
![image](https://user-images.githubusercontent.com/26747447/79056368-28045800-7c2c-11ea-9d4b-1b9d1dfb1757.png)

**Comment by Xaseron** [comment]
![image](https://user-images.githubusercontent.com/1352894/79078298-c9141100-7d07-11ea-9d6b-f48596ee4d71.png)

This is the maxium size of the PIP on my screen and i think it could be larger. Can you please remove the maxium PIP size restriction?

**Comment by petternilsen** [comment]
That size restriction is in Chromium and not something the extension can control.

**Comment by djbclark** [comment]
> That size restriction is in Chromium and not something the extension can control.

If it's easy, could you tell us where this size restriction in Chromium comes from (e.g. what the variable name is)? I'd also love to have a size larger than ~1/4 of the screen, and would be willing to go open an issue request against Chromium itself. 

Use case: A lot of board games have a clock / move list that you don't really need to pay attention to; so it'd be useful to have the window float on top of that.

**Comment by petternilsen** [comment]
The extensions popup is not controlling the size of the PIP window, that code is in overlay_window_views.cc.  OverlayWindowViews::UpdateMaxSize() set the max size to half the size of the active monitor, on both axis:

  max_size_ = gfx::Size(work_area.width() / 2, work_area.height() / 2);

**Comment by vchirikov** [comment]
It's really too small, Firefox has a larger PiP window and this is really useful for some cases.

**Comment by surajbahadur591** [comment]
anyone found the solution to make pip screen bigger?

**Comment by kened-ferreira** [comment]
Try using Opera. It doesn't have this limit.

**Comment by kened-ferreira** [comment]
And Opera pip is more complete. It has pip progress bar and volume control, using mouse scroll.

**Comment by heitorgiacominibrasil** [comment]
Use firefox, it have an native picture-in-picture function and have no max-size limit.

**Comment by surajbahadur591** [comment]
use firefox or opera for maximum window size in pip, in chrome pip size is limited and I found no way to increase

**Comment by qwabra** [comment]
> It's really too small, Firefox has a larger PiP window and this is really useful for some cases.

just gone out to FF, thanks a lot.

**Comment by kibblewhite** [comment]
Experiencing the small issue, why is there a limitation on the size that picture-in-picture can go up to? Really odd... I would like to make it the width of the screen at least

**Comment by vincentcsm0303** [comment]
Yeah.. the limitation very annoying to me, why there's a limit there.

**Comment by juliandreas** [comment]
Both Opera and Brave, which are Chromium based, have managed to figure this out

**Comment by Technetium1** [comment]
Unless someone contributes it, Google doesn't give a half of a damn to do it. I've been waiting for this for more than 2 years now.

**Comment by kibblewhite** [comment]
Maybe if enough of us start commenting on their youtube channels too, it might bring some attention or at least a response?

**Comment by Technetium1** [comment]
@beaufortfrancois is the last one that interacted with this repo that has any power or _might_ care.

**Comment by Technetium1** [comment]
Only 4 more days until we've all been waiting a total of 800 days.

**Comment by Garulf** [comment]
This is *very* limiting and I see not benefit in having a max size like this. Please consider removing the limit.

**Comment by atagulalan** [comment]
> The extensions popup is not controlling the size of the PIP window, that code is in overlay_window_views.cc. OverlayWindowViews::UpdateMaxSize() set the max size to half the size of the active monitor, on both axis:
> 
> max_size_ = gfx::Size(work_area.width() / 2, work_area.height() / 2);

This is really bad. I have two vertical monitors and max width is limited to 540px. I know this is not a priority but is it really time-consuming to change one line? Remove 8 characters from this line and you are good to go.

`max_size_ = gfx::Size(work_area.width(), work_area.height());`

Come on now.

**Comment by petternilsen** [comment]
This extension does not control this size. File a bug with chromium or use Vivaldi.

**Comment by kibblewhite** [comment]
I guess one would report it here?
https://bugs.chromium.org/p/chromium/issues/entry?template=Audio%2FVideo+Issue

**Comment by beaufortfrancois** [comment]
I've just updated Chromium so that the Picture-in-Picture window has a max size of
80% of the display width/height, not 50% anymore. See https://chromiumdash.appspot.com/commit/67df4c338d22f5d9f3e16d7a9e78187eba17fcbb

You can try in [Chrome Canary](https://www.google.com/chrome/canary/)

**Comment by debpalash** [comment]
> I've just updated Chromium so that the Picture-in-Picture window has a max size of 80% of the display width/height, not 50% anymore. See https://chromiumdash.appspot.com/commit/67df4c338d22f5d9f3e16d7a9e78187eba17fcbb
> 
> You can try in [Chrome Canary](https://www.google.com/chrome/canary/)

Any eta for stable release version?

**Comment by beaufortfrancois** [comment]
According to https://chromiumdash.appspot.com/schedule, Chrome 108 will be stable Tue, Nov 29, 2022.

**Comment by gaveitatry2** [comment]
> I've just updated Chromium so that the Picture-in-Picture window has a max size of 80% of the display width/height, not 50% anymore. See https://chromiumdash.appspot.com/commit/67df4c338d22f5d9f3e16d7a9e78187eba17fcbb
> 
> You can try in [Chrome Canary](https://www.google.com/chrome/canary/)

Can you make it so that max size is 100% or whatever we want to make it, instead of limiting us to 80%?  I want my Picture in Picture to be 1100x870 at 225, 0 on a 1600x900 desktop resolution.  But I probably can't do that with a 80% of display limit.

**Comment by ZerOmNia** [comment]
Really sad to see that this managed to get changed, but still isn't just 100%. Portrait style monitors should be able to size up to 100% width, which would still leave about 2/3 of the display available.

**Comment by gaveitatry2** [comment]
> According to https://chromiumdash.appspot.com/schedule, Chrome 108 will be stable Tue, Nov 29, 2022.

Please let us do 100% width/height, instead of 80%. I'd like to have a Picture in picture be 1100x870 at position 225,0, but with 80% width/height, I will only be able to have 1100x720 (which isn't at all good for me). I also wanted to toggle to 1600x900 sometimes (same as my desktop resolution) sometimes without leaving Picture in picture, something that can't be done with a limit of 80% width/height. And as someone mentioned, not all monitors have the same dimensions. Apparently there is a portrait style monitor that is super tall and narrow so a 100% width would still leave 2/3 of the display available. But I really don't get the limitations. If someone wants both 100% width AND 100% height of their display, they should be allowed to have that as that is their preference and helps the achieve the functionality they need to be happier and more productive.

Also, I don't know if this extension can fix this problem or not. But if I made a Picture in picture at size 260x146 and put it at 1350, 0 after a while it auto moves to position 1320, 0. And then I have to move it again. Any idea why Google does that or how it can be made to stop?

**Comment by gaveitatry2** [comment]
> I've just updated Chromium so that the Picture-in-Picture window has a max size of 80% of the display width/height, not 50% anymore. See https://chromiumdash.appspot.com/commit/67df4c338d22f5d9f3e16d7a9e78187eba17fcbb
> 
> You can try in [Chrome Canary](https://www.google.com/chrome/canary/)

Also, this is slightly different.  But I wanted a Picture in picture to be 250x146 at position 1350, 0. The minimum size for PnP's seems to be width 260px, and then Google keeps moving the Pnp to position 1320, 0 because it doesn't like Pnp's to be off screen. What can I do, or is there a change that can be made to the extension, so that Pnp's can be 250px wide (no max limit of 260px)? Basically how to lower the PnP minimum size limit?

**Comment by kibblewhite** [comment]
Can you make it so that the max size is 100%, instead of 80%, many thanks :)
